smartapps/bangali/rooms-child-app.src/rooms-child-app.groovy

```diff
@@ -2475,83 +2619,24 @@ def processCoolHeat()       {
                 else if (thisRule.toTimeType == timeSunset())
                     sunsetTimeWithOff = (thisRule.toTimeOffset ? new Date(sunsetTime.getTime() + (thisRule.toTimeOffset * 60000L)) : sunsetTime)
                 def tTime = ( thisRule.toTimeType == timeSunrise() ? sunriseTimeWithOff : ( thisRule.toTimeType == timeSunset() ? sunsetTimeWithOff : timeToday(thisRule.toTime, location.timeZone)))
-//                ifDebug("ruleNo: $ruleNo | fTime: $fTime | tTime: $tTime | nowDate: $nowDate | timeOfDayIsBetween: ${timeOfDayIsBetween(fTime, tTime, nowDate, location.timeZone)}")
+                //                ifDebug("ruleNo: $ruleNo | fTime: $fTime | tTime: $tTime | nowDate: $nowDate | timeOfDayIsBetween: ${timeOfDayIsBetween(fTime, tTime, nowDate, location.timeZone)}")
                 if (!(timeOfDayIsBetween(fTime, tTime, nowDate, location.timeZone)))    continue;
                 if (!timedRulesOnly)    {
-                    turnOn = null
+                    result = null
                     timedRulesOnly = true
                     i = 0
                     continue
                 }
                 ruleHasTime = true
             }
-//            ifDebug("ruleNo: $thisRule.ruleNo | thisRule.luxThreshold: $thisRule.luxThreshold | turnOn: $turnOn | previousRuleLux: $previousRuleLux")
-//            ifDebug("timedRulesOnly: $timedRulesOnly | ruleHasTime: $ruleHasTime")
+            ifDebug("${i} ${thisRule.ruleNo}")
+            //            ifDebug("ruleNo: $thisRule.ruleNo | thisRule.luxThreshold: $thisRule.luxThreshold | turnOn: $turnOn | previousRuleLux: $previousRuleLux")
+            //            ifDebug("timedRulesOnly: $timedRulesOnly | ruleHasTime: $ruleHasTime")
             if (timedRulesOnly && !ruleHasTime)     continue;
-            turnOn = thisRule.ruleNo
-        }
-    }
-    ifDebug("processCoolHeat: rule: $turnOn")
-    if (turnOn)     {
-        thisRule = getRule(turnOn, 't')
-        def tempRange = thisRule.tempRange
-        if (['1', '3'].contains(maintainRoomTemp))      {
-            def coolHigh = thisRule.coolTemp + (tempRange / 2f).round(1)
-            def coolLow = thisRule.coolTemp - (tempRange / 2f).round(1)
-            if (temperature >= coolHigh)     {
-                if (useThermostat)      {
-                    roomThermostat.setCoolingSetpoint(thisRule.coolTemp - thermoToTempSensor)
-                    roomThermostat.setThermostatFanMode()
-                    roomThermostat.cool()
-                }
-                else    {
-                    if (roomCoolSwitch.currentValue("switch") == 'off')     {
-                        roomCoolSwitch.on()
-                        updateMaintainIndP(roomCoolTemp)
-                        updateMaintainIndicator = false
-                    }
-                }
-            }
-            else if (temperature <= coolLow)
-                (useThermostat ? roomThermostat.off() : roomCoolSwitch.off())
```

Comment:
Explicitly not turning thermostats off to prevent unintentional freezes

Reply:
saw your notes up top on turning off thermostat being possibly catastrophic. could you expand on that?

thanks.

Reply:
If we turn the thermostat off, and ST gets disconnected or something else happens, the thermostat could stay off - if unnoticed that could lead to pipes freezing etc. Turning off via lowering the setpoint instead is safe and works well.

Reply:
ahh. you can probably tell i live in california 😄

Reply:
haha! Ahh, I wish I could say the same...